### PR TITLE
Add github action checks, and conditionals for multi-vpc support

### PR DESCRIPTION
**Pull request by ml-holdenporter**
# Pull Request

## What does this PR do?
Adds:
- Github actions for checking validation and whether formatting is needed. 
- Github action to generate the readme based on resource changes (ignores requirements)
- Permissions Boundary functionality 
- Clean up of variables
- Conditionals for allowing the usage of an ALB VPC and ECS VPC, this determines changes to the alb and security groups.
- SSM functionality

## Why is this PR being done?
This will allow for creating the ECS cluster and ALB in separate VPCs, and add github actions checks. 

## Checklist - These are **not** mandatory

- [x] I have updated the README.md if there are new procedures or changes.
- [x] I have deployed this change in another project successfully.

### CODE REVIEW

File: .github/workflows/terraform-docs.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: terraform-docs
+run-name: generate terraform-docs ${{ github.event.pull_request.title }}
+on:
+  pull_request:
+    types: [ synchronize ]
```
Comment (80bower) [inline .github/workflows/terraform-docs.yaml:5]: Why only `synchronize`?

File: .github/workflows/terraform-fmt.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: terraform-fmt
+run-name: terraform-fmt check ${{ github.event.pull_request.title }}
+on:
+  pull_request:
+    types: [ edited, assigned, opened, reopened, synchronize, ready_for_review ]
```
Comment (80bower) [inline .github/workflows/terraform-fmt.yaml:5]: Should we just use `opened`, `synchronize`, and `reopened` for these?

File: .github/workflows/terraform-validate.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+name: terraform-validate
+run-name: terraform-validate check ${{ github.event.pull_request.title }}
+on:
+  pull_request:
+    types: [ edited, assigned, opened, reopened, synchronize, ready_for_review ]
```
Comment (80bower) [inline .github/workflows/terraform-validate.yaml:5]: Same as above

File: variables.tf [hunk 4]
```diff
@@ -166,11 +188,23 @@ variable "alb_internal" {
   default     = false
 }
 
+variable "alb_vpc_id" {
+  type        = string
+  description = "The ID of ALB VPC to create ALB resources in. Defaults to the ID of ecs_vpc_id if none specified."
+  default     = ""
```
Comment (80bower) [inline variables.tf:194]: Can we just omit this default altogether or does it have to be an empty string?

**Review by 80bower — CHANGES_REQUESTED** [review]

File: variables.tf (hunk 4 above)
Comment (ml-holdenporter) [reply variables.tf:194]: This requires an empty string for proper conditional evaluation

File: .github/workflows/terraform-docs.yaml (hunk 1 above)
Comment (ml-holdenporter) [reply .github/workflows/terraform-docs.yaml:5]: My thinking was that unlike fmt or validate whose requirements for passing may change upon reopening etc, terraform-docs only ever needs to be ran if there has been a commit to this branch.

File: .github/workflows/terraform-fmt.yaml (hunk 2 above)
Comment (ml-holdenporter) [reply .github/workflows/terraform-fmt.yaml:5]: I think `ready_for_review` makes sense (or as about as much as `reopened` does), but dropping `edited` and `assigned` should be OK.

File: .github/workflows/terraform-docs.yaml (hunk 1 above)
Comment (80bower) [reply .github/workflows/terraform-docs.yaml:5]: Is that what `synchronize` is?  I thought it was when you refresh from another branch (like update changes from `main`).

I suppose I should look it up.

Comment (80bower) [reply .github/workflows/terraform-docs.yaml:5]: https://github.com/orgs/community/discussions/24567

File: .github/workflows/terraform-fmt.yaml (hunk 2 above)
Comment (80bower) [reply .github/workflows/terraform-fmt.yaml:5]: `opened` will run the changes.  This also happens if it is opened as a draft.

`ready_for_review` will trigger again, regardless of any changes, when the draft is set to ready.  Not sure if that gets us any value.

I feel like the three others will get most of the times we actually make a change.  May be worth testing though.

**Review by 80bower — APPROVED** [review]

**Pull request merged**